src/supertux/main.cpp

```diff
@@ -366,8 +366,10 @@ Main::launch_game()
 
   timelog("audio");
   SoundManager sound_manager;
-  sound_manager.enable_sound(g_config->sound_enabled);
-  sound_manager.enable_music(g_config->music_enabled);
+  sound_manager.enable_sound(g_config->sound_volume > 0 ? true : false);
+  sound_manager.enable_music(g_config->music_volume > 0 ? true : false);
```

Comment:
Then this is a fault in OpenAL, and indeed, we shouldn't waste resources. However allow the user to quickly toggle music and/or sound on/off by keeping the old button. Use OR here.